src/software/simulated_tests/validation.py

```diff
@@ -0,0 +1,276 @@
+    validation_proto_set = ValidationProtoSet()
+
+    def create_validation_proto_helper(validation):
+        """Helper function that computes the status and creates a
+        validation_proto, and updates it in the validation_proto_set.
+
+        :param validation: The validation to put into the proto
+
+        """
+        # Stores the validation result
+        validation_proto = ValidationProto()
+
+        # Get status
+        status = validation.get_validation_status(vision)
+
+        # Create validation proto
+        validation_proto.status = status
+        validation_proto.failure_msg = str(validation) + " failed"
+        validation_proto.validation_type = validation.get_validation_type()
+        validation_proto.geometry.CopyFrom(validation.get_validation_geometry(vision))
+
+        validation_proto_set.validations.append(validation_proto)
+
+        return status
+
+    # Validate the eventually validations. Eventually valids
+    for validation_sequence in list(eventually_validation_sequence_set):
+        for validation in validation_sequence:
+
+            # Add to validation_proto_set and get status
+            status = create_validation_proto_helper(validation)
+
+            # If the current validation is pending, we don't care about
+            # the next one. Keep evaluating until this one passes.
+            if status == ValidationStatus.FAILING:
+                break
+
+            # If the validation has passed, remove it from the set.
+            if status == ValidationStatus.PASSING:
+                validation_sequence.remove(validation)
+                continue
+
+    # Validate the always validations. We need to look at all of them
+    for validation_sequence in always_validation_sequence_set:
+        for validation in validation_sequence:
+            create_validation_proto_helper(validation)
+
+    return validation_proto_set
+
+
+def check_always_validation(validation_proto_set):
+    """Check always validations and make sure its always true
+
+    :param validation_proto_set: Validation proto set
+    :raises: AssertionError
+
+    """
+    for validation_proto in validation_proto_set.validations:
+        if validation_proto.validation_type == ValidationType.ALWAYS:
+            if validation_proto.status == ValidationStatus.FAILING:
+                raise AssertionError(validation_proto.failure_msg)
+
+
+def check_eventually_validation(validation_proto_set):
+    """Check eventually validations and make sure they are all true
+
+    :param validation_proto_set: Validation proto set
+    :raises: AssertionError
+
+    """
+    for validation_proto in validation_proto_set.validations:
+        if validation_proto.validation_type == ValidationType.EVENTUALLY:
+            if validation_proto.status == ValidationStatus.FAILING:
+                raise AssertionError(validation_proto.failure_msg)
+
+
+def create_validation_geometry(geometry=[]) -> ValidationGeometry:
```

Comment:
I don't think the documentation is totally clear on what ValidationGeometry is.

I'm guessing it's just the visual representations of the target regions?